middleware/queue/source/forward/handle.cpp

```diff
@@ -453,6 +453,9 @@ namespace casual
 
                         comply::to::state( state);
 
+                        if( state.runlevel < decltype( state.runlevel())::running)
+                           state.runlevel = decltype( state.runlevel())::running;
+
```

Comment:
You could just assign it to `::running`. The (only forward) state machine will only assign if the 'state' is lower/less. Hence it can only move forward through the "states".